packages/models-library/src/models_library/rabbit.py

```diff
@@ -0,0 +1,44 @@
+from typing import Dict, Optional
+
+from pydantic import BaseSettings, Extra
+from pydantic.networks import AnyUrl
+from pydantic.types import PositiveInt, SecretStr
+
+
+class RabbitDsn(AnyUrl):
+    allowed_schemes = {"amqp"}
+
+
+class RabbitConfig(BaseSettings):
+    dsn: Optional[RabbitDsn] = None
+
+    # host
+    host: str = "rabbit"
+    port: PositiveInt = 5672
+
+    # auth
+    user: str = "simcore"
+    password: SecretStr = SecretStr("simcore")
+
+    # channels
+    channels: Dict[str, str] = {
+        "log": "comp.backend.channels.log",
+        "instrumentation": "comp.backend.channels.instrumentation",
+    }
+
+    @property
+    def rabbit_dsn(self):
+        if self.dsn:
+            return str(self.dsn)
+
+        return RabbitDsn.build(
```

Comment:
MINOR: ideally these settings can be field even directly using dsn or providing host, port, user, password right?
Now, we should use validation mechanism to fill the one option when the other is provided.

See in [this example](https://github.com/ITISFoundation/osparc-simcore/blob/f675b36e0fefe31964b2295e50f832bd4691470e/services/director-v2/src/simcore_service_director_v2/core/settings.py#L136) how dsn is obtained from the other parameters. NOTICE that the ``dsn`` field is LAST. The order of fields in pydantic matter!

Reply:
ok nice.. just that pylint does not like it...

Reply:
done